File: CryptoBridge-python/src/domain/entities/ticker.py
from dataclasses import dataclass
from typing import Optional


@dataclass
class Ticker:
    symbol: str
    buy: float
    sell: float
    vol: float
    low: float
    high: float
    last: float

    def __post_init__(self):
        if not self.symbol or not isinstance(self.symbol, str):
            raise ValueError("Symbol must be a non-empty string")

        for field in ['buy', 'sell', 'vol', 'low', 'high', 'last']:
            value = getattr(self, field)
            if not isinstance(value, (int, float)) or value < 0:
                setattr(self, field, 0.0)

File: CryptoBridge-node/src/domain/entities/Ticker.js
class Ticker {
    constructor(symbol, buy, sell, vol, low, high, last) {
        if (!symbol || typeof symbol !== 'string' || symbol.trim() === '') {
            throw new Error('Symbol must be a non-empty string');
        }

        this.symbol = symbol;
        this.buy = Math.max(0, parseFloat(buy) || 0);
        this.sell = Math.max(0, parseFloat(sell) || 0);
        this.vol = Math.max(0, parseFloat(vol) || 0);
        this.low = Math.max(0, parseFloat(low) || 0);
        this.high = Math.max(0, parseFloat(high) || 0);
        this.last = Math.max(0, parseFloat(last) || 0);
    }
}

module.exports = Ticker;